src/spikeinterface/extractors/herdingspikesextractors.py

```diff
@@ -43,28 +43,39 @@ def __init__(self, file_path, load_unit_info=True):
         spike_ids = self._rf["cluster_id"][()]
         unit_ids = np.unique(spike_ids)
         spike_times = self._rf["times"][()]
+        unit_locs = self._rf["centres"][()]
 
-        if load_unit_info:
-            self.load_unit_info()
+        # if load_unit_info:
+        #     self.load_unit_info()
 
         BaseSorting.__init__(self, sampling_frequency, unit_ids)
-        self.add_sorting_segment(HerdingspikesSortingSegment(unit_ids, spike_times, spike_ids))
+        self.add_sorting_segment(HerdingspikesSortingSegment(unit_ids, spike_times, spike_ids, unit_locs))
         self._kwargs = {"file_path": str(Path(file_path).absolute()), "load_unit_info": load_unit_info}
 
         self.extra_requirements.append("h5py")
 
-    def load_unit_info(self):
-        # TODO
+    def get_unit_location(
+        self,
+        unit_id,
+        segment_index=None,
```

Comment:
why do we need segment_index here, does the units chnage over segment ?

Reply:
I don't work with multi-segmented recordings so I don't have much perspective on this. Since `get_unit_spike_train` is implemented at both the sorting extractor and sorting segment level I did the same here just in case.